widgetssdk/src/main/java/com/glia/widgets/engagement/EngagementRepository.kt

```diff
@@ -47,7 +47,10 @@ internal interface EngagementRepository {
     fun initialize()
     fun reset()
     fun resetQueueing()
-    fun endEngagement(endedBy: EndedBy)
+    //used for ending engagement after visitor ends engagement explicitly
+    fun endEngagement()
+    //used for ending engagement after integrator ends the engagement from outside of SDK, for example [GliaWidgets.endEngagement]
+    fun endEngagementSilently()
```

Comment:
Regarding the function name. I'm thinking if we can find something more meaningful instead of `Silently`? Do I understand correctly that this function is intended to be called after an engagement is actually ended and Widgets SDK just needs to update UI, etc? If so, maybe an alternative name could be `onEngagementEnded()` or whatnot? 🤔

Reply:
Ah, you see - I misunderstood it looking at its name... 🤔  I would expect something like `forceEndEngagement()`, not sure if it's only my brain's problem. Maybe some additional comment with this info might be useful.

Reply:
🤔

Reply:
`terminateEngagement()`?

Reply:
I'm kay with `terminateEngagement`.

Having looked at the edited comments, I've got an alternative idea - `endEngagementByVisitor()` and `endEngagementByIntegrator()`.

But it's up to you. I just shared an idea, you can ignore it.

Reply:
I'd prefer `terminateEngagement` because I think it describes the nature of the function. `endEngagementByVisitor()` and `endEngagementByIntegrator()` are more about the reason for ending the engagement.

Reply:
Okay. 👍